File: 485-max-consecutive-ones/485-max-consecutive-ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        reset = True
        count = 0
        max_ = 0
        
        for i in range(len(nums)):
            if nums[i] == 0:
                reset = True
                max_ = max(max_,count)
                count = 0
            else:
                if reset:
                    count = 1
                else:
                    count += 1
                reset = False
                
                
        return max(max_,count)
        

File: 485-max-consecutive-ones/485-max-consecutive-ones.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function(nums) {
    let max_cnt = 0;
    let cnt = 0;
    
    for (let i = 0;i < nums.length;i++){
        if (nums[i] === 0){
            cnt = 0;
            continue;
        }
        cnt++;
        max_cnt = Math.max(max_cnt,cnt)
    }
    
    return max_cnt
};